support/routes/v2/isobot/ops/botService.ts

```diff
@@ -0,0 +1,97 @@
+import { createHmac, timingSafeEqual } from "node:crypto"
+
+import qs from "qs"
+
+import { config } from "@config/config"
+
+import logger from "@root/logger/logger"
+import WhitelistService from "@root/services/identity/WhitelistService"
+
+export interface SlackPayload {
+  token: string
+  team_id: string
+  team_domain: string
+  channel_id: string
+  channel_name: string
+  user_id: string
+  user_name: string
+  command: string
+  text: string
+}
+
+class BotService {
+  whitelistService: WhitelistService
+
+  constructor(whitelistService: WhitelistService) {
+    this.whitelistService = whitelistService
+  }
+
+  public verifySignature(
+    signature: string,
+    timestamp: string,
+    payload: SlackPayload
+  ) {
+    // Verify timestamp is within 5 minutes
+    const currentTime = new Date().getTime() / 1000
+    const slackTimestamp = parseInt(timestamp)
+
+    if (Math.abs(currentTime - slackTimestamp) > 300) {
+      return false
+    }
+
+    // Create HMAC with the signing secret and the timestamp
+    const signingSecret = config.get("slackbot.secret")
```

Comment:
hmm why not use [bolt](https://github.com/slackapi/bolt-js/blob/main/src/receivers/verify-request.ts#L46)? rather than re-implement this logic?

Reply:
resolving, gonna swap in #1391